frontend/src/js/components/Inspector.jsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import Actions from '../actions/Actions.js'
+
+var Inspector = React.createClass({
+    getPlotItems(cur_plot_index) {
+        var items = [];
+        for (var i = 0; i < this.props.selected_cell.plots.length; i++) {
+            items.push(
+                <li key={i} className={'inspector-dropdown-item ' + (i === cur_plot_index
+                    ? 'active'
+                    : '')}>{i}</li>
+            )
+        }
+        return items;
+    },
+    render() {
+        var variable1 = '';
+        var variable2 = '';
+        var template = '';
+        var graphics_method_parent = '';
+        var graphics_method = '';
+        var cur_plot = this.props.selected_cell.plots[this.props.plot_being_edited];
+
+        if (this.props.selected_cell_row !== -1 || this.props.selected_cell_col !== -1) {
+            if (cur_plot.variables[0]) {
+                variable1 = cur_plot.variables[0];
+            }
+            if (cur_plot.variables[1]) {
+                variable2 = cur_plot.variables[1];
+            }
+            template = cur_plot.template;
+            graphics_method_parent = cur_plot.graphics_method_parent;
+            graphics_method = cur_plot.graphics_method;
+        }
+
+        return (
+            <div id='inspector' className=' scroll-area-list-parent right-side-list'>
```

Comment:
This is kind of a lot of the HTML. Shouldn't we break this up into several smaller components and conglomerate them in a container?